src/analyze/analyze-verify-util.c

```diff
@@ -72,26 +72,69 @@ int verify_prepare_filename(const char *filename, char **ret) {
         return 0;
 }
 
+static int find_unit_directory(const char *p, char **ret) {
+        _cleanup_free_ char *a = NULL, *u = NULL, *t = NULL, *d = NULL;
+        int r;
+
+        assert(p);
+        assert(ret);
+
+        r = path_make_absolute_cwd(p, &a);
+        if (r < 0)
+                return r;
+
+        r = RET_NERRNO(access(a, F_OK));
+        if (r >= 0) {
+                r = path_extract_directory(a, &d);
+                if (r < 0)
+                        return r;
+
+                *ret = TAKE_PTR(d);
+                return 0;
+        }
+
+        r = path_extract_filename(a, &u);
+        if (r < 0)
+                return r;
+
+        if (!unit_name_is_valid(u, UNIT_NAME_INSTANCE))
+                return -ENOENT;
+
+        /* If the specified unit is an instance of a template unit, then let's try to find the template unit. */
+        r = unit_name_template(u, &t);
+        if (r < 0)
+                return r;
+
+        r = path_extract_directory(a, &d);
+        if (r < 0)
+                return r;
+
+        a = mfree(a);
+        a = path_join(d, t);
+        if (!a)
+                return -ENOMEM;
+
+        if (access(a, F_OK) < 0)
+                return -errno;
```

Comment:
Above you used `RET_NERRNO`, but we don't here. This could be:
```suggestion
        r = RET_NERRNO(access(a, F_OK));
        if (r < 0)
                return r;
```

for consistency. Alternatively, I think just `return negative_errno()` works, too. I don't know how important it is to use these wrappers or not (the comment in `negative_errno()` talks about helping the compiler realize that `errno` won't ever be `0` here).

Reply:
RET_NERRNO is not necessary here (and also above). I dropped the first RET_NERRNO.